kong/tools/redis/schema.lua

```diff
@@ -31,9 +31,6 @@ return {
                 default = false
                 } },
             { server_name = typedefs.sni { required = false } }
-        },
-        entity_checks = {
-            { mutually_required = { "host", "port" }, },
```

Comment:
Once the default port was added this entity check did not make more sense. The error message when both host and port were set like:
```lua
{
  config = {
    redis = { 
      host = nil,
      port = nil,
    }
  }
}
```

said: `["@entity"] = { "all or none of these fields must be set: 'host', 'port'" }` because port was populated using default value. 
So I marked `host` as required since it's practically what needs to be set. (btw: it does not accept empty value either due to validator in `typedefs.host`)

Reply:
For the record, a `if match host then port is required`  would've solves this, but I agree with your sentiment that the base schema should be lenient which allows individual plugins to be more restrictive if desired.